mempool/src/core_mempool/index.rs

```diff
@@ -84,7 +84,12 @@ impl PartialOrd for OrderedQueueKey {
 
 impl Ord for OrderedQueueKey {
     fn cmp(&self, other: &OrderedQueueKey) -> Ordering {
-        match self.is_governance_txn.cmp(&other.is_governance_txn) {
+        match self
+            .governance_role
+            .as_ref()
+            .map(|x| x.priority())
+            .cmp(&other.governance_role.as_ref().map(|x| x.priority()))
```

Comment:
Will this do the right thing when one of the `governance_role`'s is `None`?

Reply:
It should. But I think it's another reason to get rid of the option :)